climpred_skeleton/comparison.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union
+
+import pkg_resources
+import xarray as xr
+import yaml
+
+from .core import Verification
+
+
+class Comparison(Verification):
+    """First step in pipeline. Broadcast your prediction and observations
+    in a certain manner.
+
+    Base Comparison class hints at which attributes are expected for each
+    sub comparison class. It also stores the factory for internal use, to
+    return the appropriate subfunction.
+    """
+
+    _comparison_method: str
+    _hindcast_comparison: bool
+    _probabilistic_comparison: bool
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset):
+        super().__init__(prediction, observation)
+
+    def get_comparison(self, method) -> 'Comparison':
+        try:
+            return comparison_dict[method](self._prediction, self._observation)
+        except KeyError:
+            raise ValueError(
+                f'{method} not valid keyword from {list(comparison_dict.keys())}'
+            )
+
+
+class EnsembleToObservation(Comparison):
+    """Example subclass for Comparison.
+
+    Stores useful information (e.g. whether this is a probabilistic comparison)
+    and houses main `broadcast` function, returning the modified prediction
+    and observation objects.
+    """
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset):
+        super().__init__(prediction, observation)
+        self._comparison_method = 'e2o'
+        self._hindcast_comparison = True
+        self._probabilistic_comparison = False
+
+    def broadcast(self) -> Union[xr.Dataset, xr.Dataset]:
+        if 'member' in self._prediction.dims:
+            prediction = self._prediction.mean('member')
+        else:
+            prediction = self._prediction
+        return prediction, self._observation
+
+
+class MemberToObservation(Comparison):
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset):
+        super().__init__(prediction, observation)
+        self._comparison_method = 'm2o'
+        self._hindcast_comparison = True
+        self._probabilistic_comparison = True
+
+    def broadcast(self) -> Union[xr.Dataset, xr.Dataset]:
+        observation = self._observation.expand_dims({'member': self._nmember})
+        observation['member'] = self._members
+        return self._prediction, observation
+
+
+class MemberToControl(Comparison):
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset):
```

Comment:
why can this not be xr.DataArray?

Reply:
I'll change that I think it can be. I was just trying out the type hinting.